### DISCUSSION TITLE
Change replication settings of s3 bucket

### DISCUSSION THREAD

**Discussion by PeterCxy** [seaweedfs/seaweedfs#2924] [Q&A]
It seems that the replication settings of a s3 bucket is stored as part of the fuse attributes and cannot be changed afterwards from shell. It would be nice if there is another shell command to reconfigure this (for new volumes in the bucket, of course).

BTW, does the setting in `fs.configure` take precedence over the one set directly in the s3 bucket?

**Comment by chrislusf** [comment]
are you using s3 or fuse?

**Comment by PeterCxy** [comment]
s3, but [from the code](https://github.com/chrislusf/seaweedfs/blob/master/weed/shell/command_s3_bucket_create.go#L61) it seems that the replication settings is stored in FuseAttributes

**Comment by PeterCxy** [comment]
Do you mean the one set in `fs.configure` takes precedence? Because the one in FuseAttributes (which is set during `s3.bucket.create`) clearly took effect here.

**Comment by PeterCxy** [comment]
All I know is that I set the `-replication` property while creating the bucket, and this value is only written to FuseAttributes, and it clearly did take effect.

**Comment by chrislusf** [comment]
Found the code here: https://github.com/chrislusf/seaweedfs/blob/master/weed/server/filer_server_handlers_write.go#L212

The priority is:
1. the replication setting in the url
2. the replication setting in the `fs.configure`
3. the replication setting configured in `s3.bucket.create`. 
4. the filer default replication setting

**Comment by PeterCxy** [comment]
@chrislusf Seems like setting a new replication value via fs.configure did not work for me. The new volumes created for the bucket still uses the replication setting from when it was created.

**Reply by PeterCxy** [reply]
Never mind, seems to be my user error...
